packages/server/src/operation-context.ts

```diff
@@ -0,0 +1,70 @@
+import { Task } from 'effection';
+import { ServerState, SimulationOptions, SimulationState, Simulator } from './interfaces';
+import { OperationContext } from './schema/context';
+import { v4 } from 'uuid';
+import { Slice } from '@effection/atom';
+import { createSimulation } from './simulation';
+
+type NewId = (() => string) | typeof v4;
+
+export function createOperationContext(atom: Slice<ServerState>, scope: Task, newid: NewId, simulators: Record<string, Simulator>): OperationContext {
+  let simulationsMap = new Map<string, Task>();
+  return {
+    atom,
+    scope,
+    async createSimulation(
+      simulator: string,
+      options: SimulationOptions,
+      debug: boolean
+    ): Promise<SimulationState> {
+      let simulationId = options.key ?? newid();
+      let existing = simulationsMap.get(simulationId);
+
+      if (!!existing) {
+        await existing.halt();
+      }
+
+      let slice = atom.slice("simulations", simulationId);
+
+      slice.set({
+        id:simulationId,
+        status: 'new',
+        simulator,
+        options,
+        services: [],
+        scenarios: {},
+        store: {},
+        debug,
+      });
+
+      let simulationTask = scope.run(function* () {
+        yield createSimulation(slice, simulators);
+
+        try {
+          yield;
+        } finally {
+          simulationsMap.delete(simulationId);
+          slice.remove();
+        }
+      });
+
+      simulationsMap.set(simulationId, simulationTask);
+
+      await scope.run(slice.filter(({ status }) => status === 'running').expect());
```

Comment:
can we put all of this in a single operation? that way, we only call one operation. I try to enter Effection world at as few points as possible, that way it makes error handling easier down the road because there is only one boundary:

```suggestion
      await scope.run(function*() {
        let simulationTask = yield scope.spawn(function* () {
          yield createSimulation(slice, simulators);

          try {
            yield;
          } finally {
            simulationsMap.delete(simulationId);
            slice.remove();
          }
        });        
        simulationsMap.set(simulationId, simulationTask);
        yield slice.filter(({ status }) => status === 'running').expect()
      });
```